pandas/core/generic.py

```diff
@@ -9948,11 +9948,24 @@ def _check_percentile(self, q):
         """
 
     @Appender(_shared_docs['pct_change'] % _shared_doc_kwargs)
-    def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None,
-                   **kwargs):
+    def pct_change(self, periods=1, fill_method=None, limit=None, freq=None,
+                   skipna=False, **kwargs):
+        if skipna and fill_method is not None:
```

Comment:
the fill_method defaults to ``pad`` so not sure how this makes sense, aside from which this is an API change.

also concerned that all other pandas methods have ``skipna=True`` as the default.

Reply:
I think from the OP, having ``skipna=True`` produces 2 while ``skipna=False`` produces 3 yes?

Reply:
@jreback I agree: all other methods have `skipna=True` as default. I just wanted to ask you if I should also set it here. Then, of course, it is an API breaking change, as `fill_method='pad'` will no longer be the default.

BTW, what is the meaning of OP? I do not understand your previous question.

Reply:
@jreback I have a question about setting `skipna=True` as default. Then, in order to use the parameter `fill_method`, the user should pass `skipna=False` explicitly, so that no ValueError is raised. For example:
```python
df.pct_change(fill_method='bfill', limit=2, skipna=False)
```
I do not see this very intuitive. So my question is: should we keep the parameters `fill_method` and `limit` in this method? Or should we better advise to use `fillna` before calling `pct_change`?
For example:
```python
df.fillna(method='bfill', limit=2).pct_change()
```

Reply:
Finally, I have set default `skipna=True` only if `fill_method` and `limit` are None.